File: algorithms/math/fibonacci.test.js
import fibonacci from './fibonacci.js'

const testFibonacci = () => {
    let num = fibonacci(1), expected = 1
    console.assert(num === expected, `expect to be: ${expected}, got instead: ${num}`)

    num = fibonacci(2), expected = 1
    console.assert(num === expected, `expect to be: ${expected}, got instead: ${num}`)

    num = fibonacci(6), expected = 8
    console.assert(num === expected, `expect to be: ${expected}, got instead: ${num}`)
}

testFibonacci()

File: algorithms/math/fibonacci.test.dart
import 'fibonacci.dart';

testFibonacci() {
  var num = fibonacci(1), expected = 1;
  if (num != expected) {
    print('expect fibonacci(1) to be: $expected, got instead: $num');
  }

  num = fibonacci(2); expected = 1;
  if (num != expected) {
    print('expect fibonacci(2) to be: $expected, got instead: $num');
  }

  num = fibonacci(6); expected = 8;
  if (num != expected) {
    print('expect fibonacci(6) to be: $expected, got instead: $num');
  }
}

main() {
  testFibonacci();
}
